Source/WebCore/accessibility/AXObjectCache.cpp

```diff
@@ -856,7 +874,8 @@ AccessibilityObject* AXObjectCache::getOrCreate(Node& node, IsPartOfRelation isP
     if (auto* renderer = node.renderer())
         return getOrCreate(*renderer);
 
-    if (!node.parentElement())
+    RefPtr composedParent = node.parentElementInComposedTree();
```

Comment:
can we make this a reference here instead of dereferencing later on?

Reply:
I don't think so, unfortunately, since this can be null. We could turn it into a `Ref` after null-checking, but there's not much difference between a `Ref` and a `RefPtr` that has been checked for null. Both hold a strong reference count and thus guarantee the pointed to object is valid for as long as the scope. No ergonomic difference either — both require usage of `->` to get at the underlying value.